### DISCUSSION TITLE
[docs] enabling decorators: metro complains about loose option

### DISCUSSION THREAD

**Discussion by reto-hoehener** [mobxjs/mobx#2970] [Q&A]
I tried to follow https://mobx.js.org/installation.html and https://mobx.js.org/enabling-decorators.html to the letter, but now the metro bundler tells me this:

transform[stderr]: [proposal-class-properties]: You are using the "loose: true" option and you are explicitly setting a value for the "setPublicClassFields" assumption. The "loose" option can cause incompatibilities with the other class features plugins, so it's recommended that you replace it with the following top-level option:
transform[stderr]: 	"assumptions": {
transform[stderr]: 		"setPublicClassFields": true,
transform[stderr]: 		"privateFieldsAsProperties": true
transform[stderr]: 	}

is that loose setting still required, what should I do? could you please update the docs?

**Comment by mweststrate** [comment]
The docs mention explicitly that it should be loose: false, not true

**Reply by reto-hoehener** [reply]
yes, and I have it set to false, but am still getting this message.

my babel.config.js:

```
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    // https://mobx.js.org/enabling-decorators.html
    "assumptions": {
      "setPublicClassFields": true,
      "privateFieldsAsProperties": true,
    },
    "plugins": [
      ["@babel/plugin-proposal-decorators", { "legacy": true }],
      ["@babel/plugin-proposal-class-properties", { "loose": false }]
      // In contrast to MobX 4/5, "loose" must be false!    ^
    ]
  };
};
```

**Reply by reto-hoehener** [reply]
if I remove the assumptions, the message goes away

**Reply by reto-hoehener** [reply]
I added that because of https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babelplugin-proposal-class-properties

![image](https://user-images.githubusercontent.com/13407181/121263045-5a075d80-c8b5-11eb-96aa-46af692bd169.png)

**Reply by mweststrate** [reply]
Never saw the assumptions concept before, probably something that was added recently to Babel, but no idea what it means :).

**Reply by reto-hoehener** [reply]
Ok thank you for your quick feedback. I will ignore the babel docs about assumptions for now, and not set any.
